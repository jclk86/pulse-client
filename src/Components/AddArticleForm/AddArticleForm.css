.AddArticleForm {
  padding-bottom: 40px;
  padding-top: 20px;
  margin-bottom: 30px;
  border-radius: 5px;
}

.AddArticleForm_lights_off {
  background-color: rgb(0, 0, 0, 0.7);
}

.AddArticleForm_lights_on {
  background-image: linear-gradient(
      rgba(220, 220, 220, 0.2),
      rgba(220, 220, 220, 0.2)
    ),
    url("../../images/main_bg.jpeg");
}

.container_AddArticleForm_header {
  text-align: center;
}

.container_AddArticleForm_title,
.container_AddArticleForm_image_url {
  width: 80%;
  margin: 20px auto;
}

.label_add_article_form {
  text-align: center;
  font-weight: bold;
}

.AddArticleForm_article_image_url,
.AddArticleForm_article_title {
  width: 100%;
  margin-top: 10px;
  margin-bottom: 20px;
  padding: 10px;
  font-size: 16px;
  border-radius: 3px;
  border: 1px solid transparent;
  border-top: none;
  border-bottom: 1px solid #ddd;
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.39), 0 -1px 1px #fff, 0 1px 0 #fff;
}

.AddArticleForm_article_content {
  font-size: 16px;
  border-radius: 3px;
  border: 1px solid transparent;
  border-top: none;
  border-bottom: 1px solid #ddd;
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.39), 0 -1px 1px #fff, 0 1px 0 #fff;
  font-size: 16px;
  height: 350px;
}

.AddArticleForm_select {
  margin: 10px auto;
  font-size: 16px;
  min-width: 200px;
  padding-left: 10px;
  border-radius: 3px;
  border: 1px solid transparent;
  border-top: none;
  border-bottom: 1px solid #ddd;
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.39), 0 -1px 1px #fff, 0 1px 0 #fff;
  font-size: 16px;
}

.AddArticleForm_article_image_url:focus,
.AddArticleForm_article_title:focus,
.AddArticleForm_select:focus,
.AddArticleForm_article_content:focus {
  outline-width: 0;
}

.container_AddArticleForm_btn {
  margin-top: 10px;
  display: flex;
}

.AddArticleForm_post_btn,
.AddArticleForm_cancel_btn {
  display: block;
  width: 20%;
  margin: 10px auto;
  border-radius: 10px;
  border: none;
  background-color: rgb(51, 160, 255);
  color: white;
  font-size: 16px;
  font-weight: bold;
}

.AddArticleForm_post_btn:hover,
.AddArticleForm_cancel_btn:hover {
  cursor: pointer;
}

.AddArticleForm_post_btn:focus,
.AddArticleForm_cancel_btn:focus {
  outline-width: 0;
}
