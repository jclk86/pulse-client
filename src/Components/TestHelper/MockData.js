import { DateFormatter } from "../TestHelper/MockData";

export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_tag: "News",
  article_id: 1,
  date_created: new Date(),
  author: {
    username: "Pop123"
  }
};
