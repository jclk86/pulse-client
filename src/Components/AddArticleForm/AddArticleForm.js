import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import {
  Form,
  FormLabel,
  Input,
  Textarea,
  Select,
  Button
} from "../Utils/Utils";
import ArticleListContext from "../../Context/ArticleListContext";
import "./AddArticleForm.css";
import TokenService from "../../Services/token-service";
import ArticleApiService from "../../Services/article-api-service";

class AddArticleForm extends Component {
  static contextType = ArticleListContext;
  constructor(props) {
    super(props);
    this.state = {
      title: {
        value: "",
        touched: false
      },
      content: {
        value: "",
        touched: false
      },
      tag: {
        value: "",
        touched: false
      }
    };
  }

  updateTitle = title => {
    this.setState({ title: { value: title, touched: true } });
  };

  updateContent = content => {
    this.setState({ content: { value: content, touched: true } });
  };

  updateTag = tag => {
    this.setState({ tag: { value: tag, touched: true } });
  };

  renderCategories = () => {
    const tags = ["News", "Interview", "Guide", "Diary", "Random", "Advice"];
    return tags.map(tag => (
      <option value={tag} key={tag}>
        {tag}
      </option>
    ));
  };

  handleSubmit = event => {
    event.preventDefault();
    const { title, content, tag } = this.state;
    // const token = TokenService.readJwtToken();
    const newArticle = {
      title: title.value,
      content: content.value,
      article_tag: tag.value
    };

    ArticleApiService.postArticle(newArticle)
      .then(this.context.addArticle(newArticle))
      .then(() => {
        this.props.history.push(`/articles`);
      });
  };

  render() {
    const { tag } = this.state;

    return (
      <Form
        className="AddArticleForm"
        onSubmit={event => this.handleSubmit(event)}
      >
        <div className="container_AddArticleForm_header">
          <h2>Create a post</h2>
          <div className="container_AddArticleForm_select">
            <Select name="tag" onChange={e => this.updateTag(e.target.value)}>
              <option key="no-val" value="">
                Select a category
              </option>
              {this.renderCategories()}
            </Select>
          </div>
        </div>
        <FormLabel
          htmlFor="AddArticleForm_article_title"
          className="label_add_article_form"
        ></FormLabel>
        <Input
          type="text"
          placeholder="Top 5 Destinations for This Summer"
          name="title"
          id="AddArticleForm_article_title"
          onChange={e => this.updateTitle(e.target.value)}
        ></Input>
        <FormLabel
          htmlFor="AddArticleForm_article_content"
          className="label_add_article_form"
        ></FormLabel>
        <div className="container_AddArticleForm_textarea">
          <Textarea
            placeholder="For the past two years I've traveled to..."
            name="content"
            id="AddArticleForm_article_content"
            onChange={e => this.updateContent(e.target.value)}
          ></Textarea>
        </div>

        <div className="container_AddArticleForm_btn">
          <Button
            type="button"
            role="button"
            // change below history redirect
            onClick={() => this.props.history.push("/articles")}
          >
            Cancel
          </Button>
          <Button type="submit">Post</Button>
        </div>
      </Form>
    );
  }
}

export default withRouter(AddArticleForm);
