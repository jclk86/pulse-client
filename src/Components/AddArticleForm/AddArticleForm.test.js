import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AddArticleForm from "./AddArticleForm";
import AuthApiService from "../../Services/auth-api-service";
import { mockUser, mockArticle } from "../TestHelper/MockData";
import sinon from "sinon";

describe(`AddArticleForm Component`, () => {
  beforeAll(() => {
    AuthApiService.postUser(mockUser).then(user => {
      AuthApiService.postLogin({
        username: user.username,
        password: user.password
      });
    });
  });

  it("renders AddArticleForm without issues", () => {
    const wrapper = shallow(<AddArticleForm></AddArticleForm>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("submits AddArticleForm", () => {
    const onSubmit = sinon.spy();
    const wrapper = mount(
      <BrowserRouter>
        <AddArticleForm onSubmit={onSubmit}></AddArticleForm>
      </BrowserRouter>
    );
    wrapper.find("Form").simulate("submit", {
      target: {
        title: mockArticle.title,
        content: mockArticle.content,
        article_tag: mockArticle.article_tag
      },
      preventDefault: () => {}
    });
  });
});
