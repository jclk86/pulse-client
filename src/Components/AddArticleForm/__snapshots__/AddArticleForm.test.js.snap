// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddArticleForm Component renders AddArticleForm without issues 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
